src/axom/core/memory_management.hpp

```diff
@@ -34,50 +41,40 @@ constexpr int INVALID_ALLOCATOR_ID = -1;
  * \param [in] resource_type the Umpire resource type
  * \return ID the id of the predefined umpire allocator.
  */
-inline int getResourceAllocatorID(
+inline int getUmpireResourceAllocatorID(
   umpire::resource::MemoryResourceType resource_type )
 {
   umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
   umpire::Allocator alloc     = rm.getAllocator( resource_type );
   return alloc.getId();
 }
 
-/*!
- * \brief Returns the umpire allocator associated with the given ID.
- * \param [in] allocatorID the ID of the allocator to get.
- */
-inline umpire::Allocator getAllocator( int allocatorID )
-{
-  return umpire::ResourceManager::getInstance().getAllocator( allocatorID );
-}
-
-/*!
- * \brief Sets the default memory space to use. Default is set to HOST
- * \param [in] allocator the umpire::Allocator to make default.
- */
-inline void setDefaultAllocator( umpire::Allocator allocator )
-{
-  umpire::ResourceManager::getInstance().setDefaultAllocator( allocator );
-}
+#endif
 
 /*!
  * \brief Sets the default memory space to use. Default is set to HOST
- * \param [in] allocatorID ID of the umpire::Allocator to use.
+ * \param [in] allocatorID ID of the allocator to use.
  */
 inline void setDefaultAllocator( int allocatorID )
 {
-  setDefaultAllocator( getAllocator( allocatorID ) );
+#ifdef AXOM_USE_UMPIRE
+  umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
+  umpire::Allocator allocator = rm.getAllocator( allocatorID );
+  rm.setDefaultAllocator( allocator );
+#endif
 }
 
 /*!
  * \brief Returns the current default memory space used.
  */
-inline umpire::Allocator getDefaultAllocator()
+inline int getDefaultAllocatorID()
 {
-  return umpire::ResourceManager::getInstance().getDefaultAllocator();
-}
-
+#ifdef AXOM_USE_UMPIRE
+  return umpire::ResourceManager::getInstance().getDefaultAllocator().getId();
+#else
+  return DEFAULT_ALLOCATOR_ID;
```

Comment:
For clarity, should we use ``axom::DEFAULT_ALLOCATOR_ID`` instead of just ``DEFAULT_ALLOCATOR_ID``?

Reply:
I agree, `axom::DEFAULT_ALLOCATOR_ID` is a better choice here.  Done.